### PR TITLE
Add `createCacheDirectory` capability.

### PR DESCRIPTION
**Pull request by rwjblue**
See https://github.com/broccolijs/broccoli-plugin/pull/19 for more details.

Requires merging and release of the following PR's:

- [x] https://github.com/broccolijs/broccoli-node-info/pull/1
- [x] https://github.com/broccolijs/broccoli-plugin/pull/19
- [x] Update documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,13 +35,14 @@
     "underscore.string": "^3.2.2"
   },
   "devDependencies": {
-    "broccoli-plugin": "^1.2.0",
+    "broccoli-plugin": "rwjblue/broccoli-plugin#add-capabilities",
```
Comment (rwjblue) [inline package.json]: Note: update this to 1.3.0 when https://github.com/broccolijs/broccoli-plugin/pull/19 is landed and released.

File: test/builder_test.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ var chaiAsPromised = require('chai-as-promised'); chai.use(chaiAsPromised)
 var sinonChai = require('sinon-chai'); chai.use(sinonChai)
 var multidepRequire = require('multidep')('test/multidep.json')
 
-var Plugin = multidepRequire('broccoli-plugin', '1.2.0')
+var Plugin = require('broccoli-plugin')
```
Comment (rwjblue) [inline test/builder_test.js]: Note: update this back to `multidepRequire('broccoli-plugin', '1.3.0') when https://github.com/broccolijs/broccoli-plugin/pull/19 lands.

File: test/builder_test.js [hunk 3]
```diff
@@ -72,6 +72,25 @@ describe('Builder', function() {
   })
 
   describe('broccoli-plugin nodes (nodeType: "transform")', function() {
+    it('does not create a cachePath when opt-ed out', function() {
```
Comment (rwjblue) [inline test/builder_test.js]: Note: move this back into the multidep require block below with a `if (semver.gt(version, '1.3.0')) {}` guard when https://github.com/broccolijs/broccoli-plugin/pull/19 lands.

File: test/multidep.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "path": "test/multidep",
   "versions": {
-    "broccoli-plugin": ["1.0.0", "1.1.0", "1.2.0"],
+    "broccoli-plugin": ["1.0.0", "1.1.0", "1.2.2"],
```
Comment (rwjblue) [inline test/multidep.json]: Note: add 1.3.0 when https://github.com/broccolijs/broccoli-plugin/pull/19 lands

**Comment by joliss** [comment]
The code looks good to me! I think it's good to merge! (Modulo your notes above, obviously. And there's a minor conflict, because we got rid of broccoli-kitchen-sink-helpers in the meantime.)

File: test/builder_test.js (hunk 3 above)
Comment (rwjblue) [inline test/builder_test.js]: Nice, that is great (I should RFTM a bit better next time 😝 ).

**Pull request merged**